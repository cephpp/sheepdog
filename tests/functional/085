#!/bin/bash

# Restart sheep with less disks in MD setup

. ./common

MD=true

for i in `seq 0 5`; do
    _start_sheep $i
done

_wait_for_sheep 6

_cluster_format -c 6

_vdi_create test 100M
# create 25 objects
for i in `seq 0 24`; do
    echo $i | $DOG vdi write test $((i * 4 * 1024 * 1024)) 512
done

$DOG vdi read test | md5sum

$DOG cluster shutdown
_wait_for_sheep_stop

for i in `seq 0 4`; do
    _start_sheep $i
done

# start sheep with different number of disks than that before shutdown
MD_STORE=",$STORE/5/d0,$STORE/5/d1,$STORE/5/d2,$STORE/5/d3"
_start_sheep 5

_wait_for_sheep 6

for i in `seq 0 24`; do
    echo $(($i+1)) | $DOG vdi write test $((i * 4 * 1024 * 1024)) 512
done
$DOG vdi read test | md5sum
$DOG cluster info | _filter_cluster_info
