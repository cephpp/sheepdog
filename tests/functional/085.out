QA output created by 085
using backend plain store
9c7766570b3be3aff2724f587c2f4107  -
a6c5d562057d0b8e7c90df17e5ea9450  -
Cluster status: running, auto-recovery enabled

Cluster created at DATE

Epoch Time           Version
DATE      1 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002, 127.0.0.1:7003, 127.0.0.1:7004, 127.0.0.1:7005]
